src/rulesets/oas/functions/__tests__/oasPathParam.test.ts

```diff
@@ -233,4 +299,171 @@ describe('oasPathParam', () => {
       },
     ]);
   });
+
+  test('Error if path parameter definition is set (at the global and/or operation level), but unused', async () => {
+    const results = await s.run({
+      paths: {
+        '/foo': {
+          parameters: [
+            {
+              name: 'boo',
+              in: 'path',
+              required: true,
+            },
+          ],
+          get: {
+            parameters: [
+              {
+                name: 'bar',
+                in: 'path',
+                required: true,
+              },
+            ],
+          },
+          put: {
+            parameters: [
+              {
+                name: 'baz',
+                in: 'path',
+                required: true,
+              },
+              {
+                name: 'qux',
+                in: 'path',
+                required: true,
+              },
+            ],
+          },
+        },
+      },
+    });
+
+    expect(results).toEqual([
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Parameter `boo` is not used in the path `/foo`.',
+        path: ['paths', '/foo', 'parameters', '0'],
+        severity: DiagnosticSeverity.Error,
+      }),
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Parameter `bar` is not used in the path `/foo`.',
+        path: ['paths', '/foo', 'get', 'parameters', '0'],
```

Comment:
@P0lip This duplicated parameter (at path and operation level) wasn't previously detected